headlamp-k8s/headlamp
# WIP: Allow plugins to register themes

This is a WIP PR for registering themes from plugins.
It's still very rough, but it's working...

![Screenshot showing the theme changed](https://github.com/user-attachments/assets/58687cb2-2c96-4a02-bbd5-aebbbb142ad6)

fixes #2150 

How to test:
1. Go to plugins/headlamp-plugin and run `npm run build`
2. Then run `npm pack`
3. Then go to the new plugins/examples/theme and install the headlamp-plugin we packed there: `npm i ../../headlamp-plugin/kinvolk-headlamp-plugin...tar.gz`
4. Run the plugin with `npm i ; npm run start`
5. Go to the Headlamp settings to see new themes

Progress:
- [ ] Add proper typing for themes
- [ ] Abstract themes into their own definition/module
- [ ] Sanitize the themes to not have stuff like `somethingBg` but also `something: { background: ... }`
- [ ] Replace the primaryColor usage by something more appropriate
- [ ] Allow plugins to register themes
- [ ] Allow to change the theme from settings
- [ ] Create a proper theme button which previews the colors for the theme
- [ ] Allow to associate a logo with the theme
